### PR TITLE
VIDEO-3720: Chat Integration Layer - Read New Messages

### PR DESCRIPTION
**Pull request by Alton09**
## Description

Completes the integration layer work by adding support for reading in new messages and handles state management regarding when messages are considered unread by the `ChatManager`.

## Validation

- Passed CI
- Manually test sending and receiving new messages

## Submission Checklist
 - [x] The source has been evaluated for semantic versioning changes and are reflected in ```versionName``` under `app/build.gradle`
 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code

### CODE REVIEW

File: app/src/community/java/com/twilio/video/app/data/api/AuthServiceRequestDTO.kt [hunk 1]
```diff
@@ -4,5 +4,6 @@ data class AuthServiceRequestDTO(
     val passcode: String? = null,
     val user_identity: String? = null,
     val room_name: String? = null,
-    val create_room: Boolean? = false
+    val create_room: Boolean? = true,
```
Comment (kbagchiGWC) [inline app/src/community/java/com/twilio/video/app/data/api/AuthServiceRequestDTO.kt:7]: The default value of `create_room: Boolean` changed to `true`. Is that intenetional?

**Review by kbagchiGWC — APPROVED** [review]

File: app/src/community/java/com/twilio/video/app/data/api/AuthServiceRequestDTO.kt [hunk 2]
```diff
@@ -4,5 +4,6 @@ data class AuthServiceRequestDTO(
...
+    val create_conversation: Boolean? = true
```
Comment (kbagchiGWC) [inline app/src/community/java/com/twilio/video/app/data/api/AuthServiceRequestDTO.kt:8]: Is this being used?

File: app/src/community/java/com/twilio/video/app/data/api/AuthServiceRequestDTO.kt (hunk 1 above)
Comment (Alton09) [reply app/src/community/java/com/twilio/video/app/data/api/AuthServiceRequestDTO.kt:7]: Good question! This DTO object can be ignored for now. On this branch, this change along with the chat UI button code are used to test out the conversation development back end quick deploy server that @timmydoza is working on along with the `ChatManagerImpl` in the app. In one of the UI stories I'll finalize this logic and add some tests behind it.

Comment (Alton09) [reply app/src/community/java/com/twilio/video/app/data/api/AuthServiceRequestDTO.kt:7]: But good call out though. In the state that this logic is in, it should not yet be merged to master until the `feature/conversations` is complete

**Pull request merged**